go/src/koding/klient/machine/mount/index/cache_test.go

```diff
@@ -0,0 +1,140 @@
+package index
+
+import (
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+)
+
+func TestCachedIndexCreate(t *testing.T) {
+	if err := cleanTempDirs(); err != nil {
+		t.Fatalf("want err = nil; got %v", err)
+	}
+	defer cleanTempDirs()
+
+	root, clean, err := generateTree()
+	if err != nil {
+		t.Fatalf("want err = nil; got %v", err)
+	}
+	defer clean()
+
+	idx, err := GetCachedIndex(root)
+	if err != nil {
+		t.Fatalf("want err = nil; got %v", err)
+	}
+
+	// Check index integrity.
+	count, diskSize, err := HeadCachedIndex(root)
+	if err != nil {
+		t.Fatalf("want err = nil; got %v", err)
+	}
+	if count != idx.Count(-1) {
```

Comment:
`if n := idx.Count(-1); n != count {`

If `Count` is buggy you have no guarantee second invocation of it will return the same value as the first one.